Observer/Frontend/CustomerRegistered.php

```diff
@@ -8,41 +8,49 @@
 use Sailthru\MageSail\Helper\Settings as SailthruSettings;
 use Sailthru\MageSail\Cookie\Hid as SailthruCookie;
 use Sailthru\MageSail\Helper\VarHelper;
+use Sailthru\MageSail\Logger;
 
 class CustomerRegistered implements ObserverInterface
 {
     private $clientManager;
     private $sailthruSettings;
     private $sailthruCookie;
     private $sailthruVars;
+    /** @var Logger */
+    private $logger;
 
     public function __construct(
         ClientManager $clientManager,
         SailthruSettings $sailthruSettings,
         SailthruCookie $sailthruCookie,
-        VarHelper $sailthruVars
+        VarHelper $sailthruVars,
+        Logger $logger
     ) {
         $this->clientManager = $clientManager;
         $this->sailthruSettings = $sailthruSettings;
         $this->sailthruCookie = $sailthruCookie;
         $this->sailthruVars = $sailthruVars;
+        $this->logger = $logger;
     }
 
     public function execute(Observer $observer)
     {
-        $requestParams = $_REQUEST;
-        $issubscribed = isset($requestParams['is_subscribed']) ? 1 : 0;
-        $optout_email = $issubscribed == 1 ? "none" : "basic";
-        
         $customer = $observer->getData('customer');
+        $isSubscribed = $observer->getEvent()->getAccountController()->getRequest()->getParam('is_subscribed');
+        if($isSubscribed == 1 || $isSubscribed == '1') {
```

Comment:
== should handle 1 or '1' scenarios automatically can you please confirm and remove the additional condition?
also convert this to ternary operator for better readability

Reply:
Done